src/app/venter/complaints-home/complaints-home.component.ts

```diff
@@ -0,0 +1,65 @@
+import { Component, OnInit } from '@angular/core';
+import { IComplaint } from '../../interfaces';
+import { DataService } from '../../data.service';
+import { API } from '../../../api';
+import { MatSnackBar } from '@angular/material/snack-bar';
+import { VenterDataService } from '../venter-data.service';
+
+@Component({
+  selector: 'app-complaints-home',
+  templateUrl: './complaints-home.component.html',
+  styleUrls: ['./complaints-home.component.css'],
+})
+export class ComplaintsHomeComponent implements OnInit {
+
+  public complaints = {} as IComplaint[];
```

Comment:
You're basically casting a JSON object to an array here, which works cause the way JS is, but this is terribly confusing. In general, it's better to initialize everything as `null` (i.e. no initialization, just specify the type) and then put `*ngIf`s to check if data has been loaded async.

Reply:
I tried that but I keep getting a TypeError (Cannot find property '...' of undefined)

Reply:
That means you're trying to access some member for the variable before it is initialized (probably a misplaced `ngIf`). Before accessing anything in the variable, you need to check if it isn't null, easiest way to do which is add a `*ngIf="complaints"` to the container in which it is being accessed.